contrib/native/client/src/clientlib/drillClientImpl.cpp

```diff
@@ -2143,6 +2146,9 @@ connectionStatus_t PooledDrillClientImpl::connect(const char* connStr, DrillUser
             Utils::shuffle(drillbits);
             // The original shuffled order is maintained if we shuffle first and then add any missing elements
             Utils::add(m_drillbits, drillbits);
+            if (m_drillbits.empty()){
+                return handleConnError(CONN_FAILURE, getMessage(ERR_CONN_ZKNODBIT));
```

Comment:
Thanks, @laurentgo! A new ticket has been filed for the suggestion: [DRILL-5335](https://issues.apache.org/jira/browse/DRILL-5335)